packages/hardhat-chai-matchers/src/internal/reverted/revertedWithoutReason.ts

```diff
@@ -1,38 +1,46 @@
+import { buildAssert } from "../../utils";
 import { decodeReturnData, getReturnDataFromError } from "./utils";
 
 export function supportRevertedWithoutReason(Assertion: Chai.AssertionStatic) {
   Assertion.addMethod("revertedWithoutReason", function (this: any) {
+    // see buildAssert's jsdoc
+    const negated = this.__flags.negate;
+
     const onSuccess = () => {
-      this.assert(
+      const assert = buildAssert(negated, onSuccess);
+
+      assert(
         false,
         `Expected transaction to be reverted without a reason, but it didn't revert`
       );
     };
 
     const onError = (error: any) => {
+      const assert = buildAssert(negated, onError);
+
       const returnData = getReturnDataFromError(error);
       const decodedReturnData = decodeReturnData(returnData);
 
       if (decodedReturnData.kind === "Error") {
-        this.assert(
+        assert(
           false,
           `Expected transaction to be reverted without a reason, but it reverted with reason '${decodedReturnData.reason}'`
         );
       } else if (decodedReturnData.kind === "Empty") {
-        this.assert(
+        assert(
           true,
-          null,
+          undefined,
```

Comment:
Just out of curiosity, why did these `null`s need to change to `undefined`s?

Reply:
IIRC, chai was expecting null here, but we tend to use `undefined` as a null value, so I did that when I wrote the `assert` helper just for consistency.